src/spxmod/regmod_builder.py

```diff
@@ -0,0 +1,246 @@
+                cmat = sp.vstack(
+                    [-cmat[neg_valid], cmat[pos_valid]], format="csr"
+                )
+                cvec = np.hstack([-cvec[0][neg_valid], cvec[1][pos_valid]])
+
+        self.cmat = asmatrix(sp.csc_matrix(cmat))
+        self.cvec = cvec
+
+    def detach_df(self) -> None:
+        self.data.detach_df()
+        del self.mat
+        del self.uvec
+        del self.gvec
+        del self.linear_uvec
+        del self.linear_gvec
+        del self.linear_umat
+        del self.linear_gmat
+        del self.cmat
+        del self.cvec
+
+    def hessian_from_gprior(self) -> Matrix:
+        """Hessian matrix from the Gaussian prior.
+
+        Returns
+        -------
+        Matrix
+            Hessian matrix.
+
+        """
+        hess = sp.diags(1.0 / self.gvec[1] ** 2, format="csr")
+        if self.linear_gvec.size > 0:
+            hess += (
+                self.linear_gmat.T.scale_cols(1.0 / self.linear_gvec[1] ** 2)
+            ).dot(self.linear_gmat)
+        return asmatrix(hess)
+
+    def fit(
+        self,
+        data: DataFrame,
+        encode: Callable,
+        optimizer: Callable = msca_optimize,
+        **optimizer_options,
+    ) -> None:
+        self.attach_df(data, encode)
+        super().fit(optimizer=optimizer, **optimizer_options)
+        self.detach_df()
+
+    def predict(
+        self,
+        data: DataFrame,
+        encode: Callable,
+        return_ui: bool = False,
+        alpha: float = 0.05,
+    ) -> NDArray:
+        mat = sp.csc_matrix(encode(data))
+        param, coef = self.params[0], self.opt_coefs
+
+        offset = np.zeros(len(data))
+        if param.offset is not None:
+            offset = data[param.offset].to_numpy()
+
+        lin_param = offset + mat.dot(coef)
+        pred = param.inv_link.fun(lin_param)
+
+        if return_ui:
+            if alpha < 0 or alpha > 0.5:
+                raise ValueError("`alpha` has to be between 0 and 0.5")
+            # TODO: explore the sparsity of the variance-covariance matrix
+            vcov = get_vcov(self.opt_hessian, self.opt_jacobian2)
+            lin_param_sd = np.sqrt(get_pred_var(mat, vcov))
+            lin_param_lower = norm.ppf(
+                0.5 * alpha, loc=lin_param, scale=lin_param_sd
+            )
+            lin_param_upper = norm.ppf(
+                1 - 0.5 * alpha, loc=lin_param, scale=lin_param_sd
+            )
+            pred = np.vstack(
+                [
+                    pred,
+                    param.inv_link.fun(lin_param_lower),
+                    param.inv_link.fun(lin_param_upper),
+                ]
+            )
+        return pred
+
+
+class SparseBinomialModel(SparseRegmodModel, BinomialModel):
+    """Sparse Binomial model."""
+
+
+class SparseGaussianModel(SparseRegmodModel, GaussianModel):
+    """Sparse Gaussian model."""
+
+
+class SparsePoissonModel(SparseRegmodModel, PoissonModel):
+    """Sparse Poisson model."""
```

Comment:
I don't have a better name, but this could be confusing later, since we're not doing sparse regression, we're just taking advantage of sparse features. Maybe something like SparseMatrixBinomialModel.

Reply:
Yes! We can think of better names later on, these are temporary placeholders, and actual change will happen in `regmod` :)